---
output: pdf_document
geometry: margin=1in
mainfont: Times New Roman
---

```{r load_data,warning=F, message=F, echo=F,comment="K",results='hide'}
#load all datasets

library(repmis)
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS1_dosedata.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS2_allRIAILregressed.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS3_allRIAILsPC.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS4_allAnnotatedLods.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS5_scantwosummary.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS6_varianceComponents.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS8_PC_trait_correlation.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS9_allNILCSSregressed.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS10_css_nil_stats.Rda?raw=TRUE")
source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS11_assays_category.Rda?raw=TRUE")
library(linkagemapping)
library(ggdendro)
library(tidyverse)
data("N2xCB4856cross")
knitr::opts_knit$set(progress = FALSE)


```


```{r s1,warning=F, message=F, echo=F,comment="K",results='asis'}
library(pander)
library(stringr)
library(knitr)
library(scales)
library(cowplot)
library(tidyverse)
library(reshape2)
library(repmis)

panderOptions('knitr.auto.asis', TRUE)

cat("\n\n\n")
cat("\n\n\\pagebreak\n")

#### Dose response
dosedrugs <- c("cadmium", "carmustine", "chlorothalonil", "chlorpyrifos", "cisplatin", "copper", "diquat", "fluoxetine", "FUdR", "irinotecan", "mechlorethamine", "paraquat", "silver", "topotecan", "tunicamycin", "vincristine")

dosecontrols <- c("water", "DMSO", "DMSO", "DMSO", "water", "water", "water", "DMSO", "water" ,"DMSO", "water", "water", "water", "water", "DMSO", "water")

doseplates <- c(23, 5, 6, 22, 3, 4, 2, 11, 9, 12, 1, 7, 21, 15, 17, 8)

cadmiumconc <- factor(c("water", "cadmium100", "cadmium200", "cadmium300", "cadmium400"), levels = c("water", "cadmium100", "cadmium200", "cadmium300", "cadmium400"))
carmustineconc <- factor(c("DMSO", "carmustine125", "carmustine250", "carmustine500","carmustine1000"), levels = c("DMSO", "carmustine125", "carmustine250", "carmustine500","carmustine1000"))
chlorothalonilconc <- factor(c("DMSO","chlorothalonil125","chlorothalonil250","chlorothalonil500","chlorothalonil1000"), levels = c("DMSO","chlorothalonil125","chlorothalonil250","chlorothalonil500","chlorothalonil1000"))
chlorpyrifosconc <- factor(c("DMSO","chlorpyrifos025","chlorpyrifos05","chlorpyrifos1","chlorpyrifos2"), levels =c("DMSO","chlorpyrifos025","chlorpyrifos05","chlorpyrifos1","chlorpyrifos2"))
cisplatinconc <- factor(c("water","cisplatin125","cisplatin250","cisplatin500","cisplatin1000"), levels = c("water","cisplatin125","cisplatin250","cisplatin500","cisplatin1000"))
copperconc <- factor(c("water","copper625","copper125","copper250","copper500"), levels = c("water","copper625","copper125","copper250","copper500"))
diquatconc <- factor(c("water","diquat250","diquat500","diquat1000","diquat2000"), levels = c("water","diquat250","diquat500","diquat1000","diquat2000"))
fluoxetineconc <- factor(c("DMSO","fluoxetine625","fluoxetine125","fluoxetine250","fluoxetine500"), levels = c("DMSO","fluoxetine625","fluoxetine125","fluoxetine250","fluoxetine500"))
fudrconc <- factor(c("water", "FUdR375", "FUdR50", "FUdR75", "FUdR100"), levels = c("water", "FUdR375", "FUdR50", "FUdR75", "FUdR100"))
irinotecanconc <- factor(c("DMSO","irinotecan625","irinotecan125","irinotecan250","irinotecan500"), levels = c("DMSO","irinotecan625","irinotecan125","irinotecan250","irinotecan500"))
mechlorethamineconc <- factor(c("water","mechlorethamine200","mechlorethamine300","mechlorethamine400","mechlorethamine500"), levels = c("water","mechlorethamine200","mechlorethamine300","mechlorethamine400","mechlorethamine500"))
paraquatconc <- factor(c("water","paraquat500","paraquat1000","paraquat2000","paraquat4000"), levels = c("water","paraquat500","paraquat1000","paraquat2000","paraquat4000"))
silverconc <- factor(c("water","silver75","silver150","silver300","silver500"), levels = c("water","silver75","silver150","silver300","silver500"))
topotecanconc <- factor(c("water","topotecan50","topotecan100","topotecan200","topotecan400"), levels = c("water","topotecan50","topotecan100","topotecan200","topotecan400"))
tunicamycinconc <- factor(c("DMSO","tunicamycin5","tunicamycin10","tunicamycin15","tunicamycin20"), levels = c("DMSO","tunicamycin5","tunicamycin10","tunicamycin15","tunicamycin20"))
vincristineconc <- factor(c("water","vincristine20","vincristine40","vincristine60","vincristine80"), levels = c("water","vincristine20","vincristine40","vincristine60","vincristine80"))


concentrations <- list(cadmiumconc, carmustineconc, chlorothalonilconc, chlorpyrifosconc, cisplatinconc, copperconc, diquatconc, fluoxetineconc, fudrconc, irinotecanconc, mechlorethamineconc, paraquatconc, silverconc, topotecanconc, tunicamycinconc, vincristineconc)

cadmiumlabs <- c("0", "100","200","300","400")
carmustinelabs <- c("0", "125", "250", "500", "1000")
chlorothalonillabs <- c("0", "125", "250", "500", "1000")
chlorpyrifoslabs <- c("0",".25",".5","1","2")
cisplatinlabs <- c("0","125","250","500","1000")
copperlabs <- c("0","62.5","125","250","500")
diquatlabs <- c("0","250","500","1000","2000")
fluoxetinelabs <- c("0","62.5","125","250","500")
fudrlabs <- c("0", "37.5", "50", "75", "100")
irinotecanlabs <- c("0","62.5","125","250","500")
mechlorethaminelabs <- c("0","200","300","400","500")
paraquatlabs <- c("0","500","1000","2000","4000")
silverlabs <- c("0","75","150","300","500")
topotecanlabs <- c("0","50","100","200","400")
tunicamycinlabs <- c("0","5","10","15","20")
vincristinelabs <- c("0","20","40","60","80")

doselabels <- list(cadmiumlabs, carmustinelabs, chlorothalonillabs, chlorpyrifoslabs, cisplatinlabs, copperlabs, diquatlabs, fluoxetinelabs, fudrlabs, irinotecanlabs, mechlorethaminelabs, paraquatlabs, silverlabs, topotecanlabs, tunicamycinlabs, vincristinelabs)

##### Trait correlation
traitcordrugs <- dosedata %>%
    dplyr::filter(condition %in% c("cadmium100", "carmustine250","chlorothalonil250","chlorpyrifos1","cisplatin250","copper250","diquat250","fluoxetine250","FUdR50","irinotecan125","mechlorethamine200","paraquat500","silver150","topotecan400","tunicamycin10","vincristine80"))

traitcordrugs$condition <- factor(traitcordrugs$condition, levels = c("cadmium100", "carmustine250","chlorothalonil250","chlorpyrifos1","cisplatin250","copper250","diquat250","fluoxetine250","FUdR50","irinotecan125","mechlorethamine200","paraquat500","silver150","topotecan400","tunicamycin10","vincristine80"), labels = c("Cadmium 100 µM", "Carmustine 250 µM","Chlorothalonil 250 µM","Chlorpyrifos 1 µM","Cisplatin 250 µM","Copper 250 µM","Diquat 250 µM","Fluoxetine 250 µM","FUdR 50 µM","Irinotecan 125 µM","Mechlorethamine 200 µM","Paraquat 500 µM","Silver 150 µM","Topotecan 400 µM","Tunicamycin 10 µM","Vincristine 80 µM"))

splitbydose <- split(traitcordrugs, paste(traitcordrugs$condition))

```

```{r s1_2, fig.width=7.5, fig.height=3,warning=F, message=F, echo=F,comment="K",results='asis', dpi = 300}

reorder_cormat <- function(cormat){
    # Use correlation between variables as distance
    dd <- as.dist((1-cormat)/2)
    hc <- hclust(dd)
    cormat <-cormat[hc$order, hc$order]
}

for (i in 1:length(dosedrugs)) {

    traitdf <- dosedata %>%
        dplyr::filter(trait %in% c("norm.n", "mean.TOF","mean.norm.EXT"), grepl(dosedrugs[i], condition)|grepl(dosecontrols[i],condition), plate == doseplates[i]) %>%
        dplyr::group_by(strain)

    traitdf$trait <- factor(traitdf$trait, levels = c("norm.n","mean.TOF","mean.norm.EXT"), labels = c("Brood Size", "Mean Length", "Mean Optical Density"))

    doseplot <- ggplot2::ggplot(traitdf) +
        ggplot2::theme_bw()+
        ggplot2::aes(x=condition, y=phenotype, fill=strain) +
        ggplot2::geom_point(position = position_jitterdodge(jitter.width = 0.2), size = .5)+
        ggplot2::geom_boxplot(outlier.size=NA, outlier.color = NA,alpha = .5)+
        ggplot2::scale_x_discrete(limits=levels(concentrations[[i]]), labels = doselabels[[i]])+
        ggplot2::scale_fill_manual(values=c("blue","green","pink","orange"), guide_legend(title = "Strain"))+
        ggplot2::ggtitle(Hmisc::capitalize(dosedrugs[i])) +
        ggplot2::labs(x = "Concentration (µM)", y = "Raw Phenotype") +
        ggplot2::facet_wrap(~ trait, scales = "free") +
        ggplot2::theme(axis.text.x = ggplot2::element_text(size=10, face = "bold", color="black"),
                       axis.text.y = ggplot2::element_text(size=10, face = "bold", color="black"),
                       axis.title.x = ggplot2::element_text(size=12, face = "bold", color="black"),
                       axis.title.y = ggplot2::element_text(size=12, face = "bold", color="black"),
                       plot.title = ggplot2::element_text(size=12, face = "bold", color = "black", hjust = .5),
                       strip.background = ggplot2::element_rect(fill = NA),
                       strip.text = ggplot2::element_text(size = 12, face = "bold", color = "black"),
                       legend.text = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                       legend.title = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                       legend.key.size = grid::unit(.2, "in"),
                       legend.margin = ggplot2::margin(0, 0, 0, 0, "in"))
    
    print(doseplot)

    cat("\n\n\n")
    cat("\n\n\n")
    cat("\n\n\n")
    # cat("\n\n\\pagebreak\n")

}
```

**Figure S1** Dose-response High Throughput Assay. (**A**) Raw phenotypic values for four divergent strains, CB4856 (blue), DL238 (green), JU258 (pink), and N2 (orange), in response to various concentrations of each toxin are plotted as Tukey box plots. Toxin responses for traits representing three distinct population parameters- brood size (norm.n), mean animal length (mean.TOF), and mean optical density (mean.norm.EXT) are shown. The x-axis indicates either the control (water or DMSO) or the concentration (µM) of the toxin, and the y-axis shows the raw phenotypic value for each trait.

```{r s2, fig.width=7.5, fig.height=8,warning=F, message=F, echo=F,comment="K",results='asis', dpi = 300}

RIAILPC <- allRIAILsPC %>%
  dplyr::mutate(drugtrait = paste(condition, trait, sep = "."))


set2RIAILs <- allRIAILregressed %>%
  dplyr::filter(grepl("QX", strain))%>%
  dplyr::mutate(strain2 = as.numeric(stringr::str_split_fixed(strain, "QX", 2)[,2])) %>%
  dplyr::filter(strain2 >239)%>%
  dplyr::select(-strain2)

reorder_cormat <- function(cormat){
  # Use correlation between variables as distance
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd) 
  cormat <-cormat[hc$order, hc$order]
}

splitbydrug <- split(set2RIAILs, set2RIAILs$condition)

for (drug in sort(unique(allRIAILsPC$condition))) {

  traitcor <- splitbydrug[[drug]] %>%
    dplyr::filter(trait %in% c("n","norm.n", "q10.TOF", "q10.EXT", "q25.TOF", "q25.EXT", "median.TOF", "median.EXT", "mean.TOF", "mean.EXT", "q75.TOF",
                               "q75.EXT", "q90.TOF", "q90.EXT", "var.TOF", "var.EXT", "cv.TOF", "cv.EXT", "q10.norm.EXT", "q25.norm.EXT", "median.norm.EXT", "mean.norm.EXT",
                               "q75.norm.EXT", "q90.norm.EXT"))
  
  
  traitcor<- traitcor %>%
    dplyr::select(-(.fitted:.std.resid)) %>%
    dplyr::group_by(trait) %>%
    dplyr::ungroup()%>%
    tidyr::spread(trait, phenotype) %>%
    dplyr::ungroup() %>%
    dplyr::select(cv.EXT:var.TOF)
  traitcormat <- stats::cor(traitcor, use = "pairwise.complete.obs")
  reordered <- reorder_cormat(traitcormat)
  
  dd <- stats::as.dist((1-traitcormat)/2)
  hc <- stats::hclust(dd)

  dend <- ggdendro::ggdendrogram(hc, labels = FALSE, leaf_labels = FALSE) +
    ggplot2::ggtitle(R.utils::capitalize(drug)) +
    ggplot2::theme(axis.title.y = ggplot2::element_blank(),
                   axis.text.x = ggplot2::element_blank(),
                   axis.text.y = ggplot2::element_blank(),
                   plot.title = ggplot2::element_text(size = 14, face = "bold", hjust = .5),
                   plot.margin = grid::unit(c(0,0,0,0), "cm"))
  
  drugpc <- RIAILPC %>%
    dplyr::filter(condition == drug)
  
  #clusters1 is the Var1 trait and which cluster it is in:
  #first divide the hc tree into the number of groups you want based on how many significant PCs we have:
  clusters1 <- data.frame(level1 = stats::cutree(hc, k = length(unique(drugpc$trait)))) %>%
    dplyr::mutate(Var1 = rownames(.))
  #then order the traits based on the way they appear in the dendrogram:
  #extract dendrogram info
  hc2 <- ggdendro::dendro_data(hc)$labels
  #factor so you can order them in the heatmap based on this order
  hc2$label <- factor(hc2$label)
  #order the traits based on their order in the dendrogram
  clusters1$Var1 <- factor(clusters1$Var1, levels = hc2$label)
  #assign cluster groups based on this order that you want the traits (they were grouped by cutree, but group number is not how we want it)
  clusters1 <- clusters1 %>%
    dplyr::arrange(Var1) %>%
    dplyr::group_by(level1)%>%
    dplyr::mutate(newlev = 1)
  for(i in 2:nrow(clusters1)){
    if(clusters1$level1[i] == clusters1$level1[i-1]){
      clusters1$newlev[i] <- clusters1$newlev[i-1]
    } else clusters1$newlev[i] <- clusters1$newlev[i-1]+1
  }
  #clusters2 is the same thing, but for clustering the y-axis traits in the heatmap
  clusters2 <- clusters1 %>%
    dplyr::rename(newlev2 = newlev, Var2 = Var1) %>%
    dplyr::ungroup()%>%
    dplyr::select(-level1)
  #add correlations and the grouping data altogether
  melted <- data.table::melt(reordered) %>%
    dplyr::left_join(clusters1)%>%
    dplyr::left_join(clusters2)
  #reorder Var2 so white line goes diagonal across whole heat map, not just within facet
  melted$Var2 <- factor(melted$Var2, levels = rev(clusters2$Var2))
  
  traitcorplot <- ggplot2::ggplot(data = melted, ggplot2::aes(x = Var1, y = Var2, fill=value^2))+
    ggplot2::geom_tile() +
    ggplot2::scale_fill_gradientn(colours=terrain.colors(6), name=expression(bold(italic("r")^2)))+
    ggplot2::scale_x_discrete(position = "top")+
    ggplot2::facet_grid(newlev2~newlev, scales = "free", space = "free", switch = "x")+
    ggplot2::theme_bw() +
    ggplot2::theme(axis.text.x = ggplot2::element_text(size=8, color="black",angle = 90, hjust=0),
          axis.text.y = ggplot2::element_text(size=8, color="black"),
          panel.background = ggplot2::element_rect(fill = '#E5E8E8'),
          strip.text = ggplot2::element_blank(),
          strip.text.y = ggplot2::element_blank(),
          panel.spacing = grid::unit(1, "pt"),
          plot.margin = grid::unit(c(-1,0,0,0),"cm"))+
    ggplot2::labs(title="", x=NULL, y=NULL) 
  
  
  finaldend <- cowplot::plot_grid(NULL, dend, NULL, nrow = 1, ncol = 3, rel_widths = c(.12, 1, .13))
  finalplot <- cowplot::plot_grid(finaldend, traitcorplot, nrow = 2, ncol = 1, rel_heights = c(.5, 1))
  
  print(finalplot)

  cat("\n\n\n")
  # don't page break on the last drug
  if(drug != "vincristine") {
        cat("\n\n\\pagebreak\n")
  }
}

```

**Figure S2** Trait correlations with RIAIL phenotypes. Correlation coefficients (r^2^) between all pairwise combinations of traits based on the RIAIL phenotypes for the indicated concentration of the toxin are plotted as a heat map. Traits are labeled on the x- and y-axes, organized by hierarchical clustering (top) of the correlation matrix (bottom) for each toxin. Colors represent correlation coefficients increasing from green to yellow to pink to white.

```{r s3,warning=F, message=F, echo=F,comment="K",results='asis'}
library(pander)
library(stringr)
library(knitr)
library(scales)
library(cowplot)
library(tidyverse)
library(reshape2)
library(linkagemapping)

panderOptions('knitr.auto.asis', TRUE)

cat("\n\n\n")
cat("\n\n\\pagebreak\n")

#Make the cross object
phenocross <- linkagemapping::mergepheno(N2xCB4856cross, allRIAILsPC, set = 2)

#VE_SE_QTL
VE <- varianceComponents %>%
    dplyr::select(condition, trait, narrowsense_h2, interaction_VE) %>%
    tidyr::gather(type, VE, narrowsense_h2:interaction_VE) %>%
    dplyr::mutate(type = ifelse(type == "narrowsense_h2", "Additive", "Interactive"))

SE <- varianceComponents %>%
    dplyr::select(condition, trait, narrowsense_h2_SE, interaction_SE) %>%
    tidyr::gather(type, SE, narrowsense_h2_SE:interaction_SE) %>%
    dplyr::mutate(type = ifelse(type == "narrowsense_h2_SE", "Additive", "Interactive"))

VE_SE_QTL <- dplyr::left_join(VE, SE) %>%
    dplyr::mutate(n = 1:nrow(.))

VE_SE_QTL$type <- factor(VE_SE_QTL$type, levels = c("Interactive", "Additive"))
VE_SE_QTL <- VE_SE_QTL %>%
    dplyr::mutate(lower = VE - SE, higher = VE + SE)

VE_SE_QTL$lower[VE_SE_QTL$type == "Interactive"] <- with(VE_SE_QTL,VE[type == "Additive"] - SE[type == "Interactive"] + VE[type == "Interactive"])
VE_SE_QTL$higher[VE_SE_QTL$type == "Interactive"] <- with(VE_SE_QTL,VE[type == "Interactive"] + VE[type == "Additive"] + SE[type == "Interactive"])

for(i in 1:nrow(VE_SE_QTL)) {
    if(VE_SE_QTL$lower[i] < 0) {
        VE_SE_QTL$lower[i] <- 0
    }
}

#Separate NIL phenotype data by experiment
chrVregressed <- allNILCSSregressed %>%
    dplyr::filter(experiment == "V")
chrIVLregressed <- allNILCSSregressed %>%
    dplyr::filter(experiment == "IVL")
chrIVRregressed <- allNILCSSregressed %>%
    dplyr::filter(experiment == "IVR")
cssVregressed <- allNILCSSregressed %>%
    dplyr::filter(experiment == "CSSV")


#Load functions
#Define function for NIL mapping
x1 <- .25
y1 <- .02
y2 <- 0.01
blck <- .12
spce <- .06
x2 <- x1+blck+spce
x3 <- x2+blck+spce
x4 <- x3+blck+spce
width <- .02
chr <- 0.02
textsize = 8
titlesize = 10
pointsize = 0.3

#Define function for NIL mapping
quick_plot_V <- function(cond, pltrt, signif, ylab = pltrt, t = "NIL"){
    phen_gen <- allNILCSSregressed %>%
        dplyr::filter(experiment == "V", trait == pltrt, condition == cond)
    
    if(is.null(signif)) {
        phen_gen$strain <- factor(phen_gen$strain, levels = unique(phen_gen$strain), ordered = T)
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = strain, y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA230" = "gray", "ECA232" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    } else {
        phen_gen$strain <- factor(phen_gen$strain, levels = unique(phen_gen$strain), ordered = T)
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = strain, y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA230" = "gray", "ECA232" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_text(data = signif, ggplot2::aes(x = strain, y = pheno, label = significant), size = 3, color = "black", fontface = "bold") +
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            #ggplot2::ylim(NA, max(phen_gen$phenotype)*1.7)+
            ggplot2::labs(y= ylab, x = "", title = t)  +
            ggplot2::facet_grid(~experiment)
    }
    
    cowplot::ggdraw() +
        cowplot::draw_plot(test, 0, .03, 1, .97) +
        cowplot::draw_label("Chr V", x = 0.01, y = .08, size = textsize, fontface = "bold", hjust = -.16)+
        ggplot2::geom_rect(ggplot2::aes(xmin = x1+chr, xmax = x1+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2+chr, xmax = x2+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+chr, xmax = x3+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+blck*.25+chr, xmax = x3+blck*.75-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+chr, xmax = x4+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+blck*.25+chr, xmax = x4+blck*.75-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        cowplot::draw_label("Genome", x = 0.01, y = .05, hjust = -.1, size = textsize, fontface = "bold") +
        ggplot2::geom_rect(ggplot2::aes(xmin = x1, xmax = x1+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2, xmax = x2+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3, xmax = x3+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4, xmax = x4+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")
}
quick_plot_IVR <- function(cond, pltrt, signif, ylab = pltrt, t = "NIL"){
    phen_gen <- allNILCSSregressed %>%
        dplyr::filter(experiment == "IVR", trait == pltrt, condition == cond)
    
    if(is.null(signif)) {
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = factor(strain, levels = c("N2", "CB4856", "ECA241", "ECA240"), 
                                    labels = c("N2", "CB4856", "ECA241", "ECA240"), ordered = T),
                         y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA241" = "gray", "ECA240" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.margin = ggplot2::margin(t = 0, r = 10, b = 30, l = 0),
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    } else{
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = factor(strain, levels = c("N2", "CB4856", "ECA241", "ECA240"),
                                    labels = c("N2", "CB4856", "ECA241", "ECA240"), ordered = T),
                         y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA241" = "gray", "ECA240" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_text(data = signif, ggplot2::aes(x = strain, y = pheno, label = significant), 
                               size = 3, color = "black", fontface = "bold") +
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            # coord_flip() +
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            # ggplot2::ylim(NA, max(phen_gen$phenotype)*1.5)+
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    }
    
    cowplot::ggdraw() +
        cowplot::draw_plot(test, 0, .03, 1, .97) +
        cowplot::draw_label("Chr IV", x = 0.01, y = .08, size = textsize, fontface = "bold", hjust = -.16)+
        ggplot2::geom_rect(ggplot2::aes(xmin = x1+chr, xmax = x1+blck-chr, ymin = y1*3, ymax = y1*3 + width),
                           colour = "black", fill = "orange")+ #N2 chr
        ggplot2::geom_rect(ggplot2::aes(xmin = x2+chr, xmax = x2+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+ #CB chr
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+chr, xmax = x3+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+ #First NIL chr
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+blck*.5+chr, xmax = x3+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+ #first NIL introgressed
        ggplot2:: geom_rect(ggplot2::aes(xmin = x4+chr, xmax = x4+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                            colour = "black", fill = "orange")+ #Last NIL chr
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+blck*.5+chr, xmax = x4+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+ #Last NIL introgressed
        cowplot::draw_label("Genome", x = 0.01, y = .05, hjust = -.1, size = textsize, fontface = "bold") +
        ggplot2::geom_rect(ggplot2::aes(xmin = x1, xmax = x1+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")+ #N2 genome
        ggplot2::geom_rect(ggplot2::aes(xmin = x2, xmax = x2+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+ #CB genome
        ggplot2::geom_rect(ggplot2::aes(xmin = x3, xmax = x3+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+ #First NIL genome
        ggplot2::geom_rect(ggplot2::aes(xmin = x4, xmax = x4+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange") #last NIL genome
}
quick_plot_cssV <- function(cond, pltrt, signif, ylab = pltrt, t = "CSS"){
    phen_gen <- allNILCSSregressed %>%
        dplyr::filter(experiment == "CSSV", trait == pltrt, condition == cond)
    
    if(is.null(signif)) {
        phen_gen$strain <- factor(phen_gen$strain, levels = unique(phen_gen$strain), ordered = T)
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = strain, y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA554" = "gray", "ECA573" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    } else {
        phen_gen$strain <- factor(phen_gen$strain, levels = unique(phen_gen$strain), ordered = T)
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = strain, y = phenotype, fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA554" = "gray", "ECA573" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_text(data = signif, ggplot2::aes(x = strain, y = pheno, label = significant), size = 3, color = "black", fontface = "bold") +
            ggplot2::geom_boxplot(outlier.colour = NA, aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           legend.position="none",
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            # ggplot2::ylim(NA, max(phen_gen$phenotype)*1.5)+
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    }
    
    cowplot::ggdraw() +
        cowplot::draw_plot(test, 0, .03, 1, .97) +
        cowplot::draw_label("Chr V", x = 0.01, y = .08, size = textsize, fontface = "bold", hjust = -.16)+
        ggplot2::geom_rect(ggplot2::aes(xmin = x1+chr, xmax = x1+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2+chr, xmax = x2+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+chr, xmax = x3+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+chr, xmax = x4+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        cowplot::draw_label("Genome", x = 0.01, y = .05, hjust = -.1, size = textsize, fontface = "bold") +
        ggplot2::geom_rect(ggplot2::aes(xmin = x1, xmax = x1+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2, xmax = x2+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3, xmax = x3+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4, xmax = x4+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")
}
quick_plot_IVL <- function(cond, pltrt, signif, ylab = pltrt, t = "NIL"){
    phen_gen <- allNILCSSregressed %>%
        dplyr::filter(experiment == "IVL", trait == pltrt, condition == cond)
    
    if(is.null(signif)) {
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = factor(strain, levels = c("N2", "CB4856", "ECA229", "ECA231"), 
                                    labels = c("N2", "CB4856", "ECA229", "ECA231"), ordered = T), 
                         y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA229" = "gray", "ECA231" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    } else {
        test <- phen_gen %>%
            ggplot2::ggplot(.)+
            ggplot2::aes(x = factor(strain, levels = c("N2", "CB4856", "ECA229", "ECA231"), 
                                    labels = c("N2", "CB4856", "ECA229", "ECA231"), ordered = T), 
                         y = phenotype,fill=strain)+
            ggplot2::scale_fill_manual(values = c("N2" = "orange", "CB4856"= "blue", "ECA229" = "gray", "ECA231" = "gray"))+
            ggplot2::geom_jitter(alpha = 1, size = pointsize)+
            ggplot2::geom_text(data = signif, ggplot2::aes(x = strain, y = pheno, label = significant), size = 3, color = "black", fontface = "bold") +
            ggplot2::geom_boxplot(outlier.colour = NA, ggplot2::aes(alpha = 0.8))+
            ggplot2::theme_bw()+
            ggplot2::theme(axis.text.x = ggplot2::element_text(size=textsize, face="bold", color="black"),
                           axis.text.y = ggplot2::element_text(size=textsize,  face = "bold", color="black"),
                           axis.title.x = ggplot2::element_text(size=titlesize, color="black", vjust=-.3, face = "bold"),
                           axis.title.y = ggplot2::element_text(size=titlesize,  color="black", face = "bold"),
                           strip.text = ggplot2::element_text(size=titlesize,  color="black"),
                           legend.position="none",
                           plot.title = ggplot2::element_text(size=titlesize, color = "black", face = "bold", hjust = 0.5),
                           panel.background = ggplot2::element_rect(color="black",size=1),
                           strip.background = ggplot2::element_rect(color = "black", size = 1)) + 
            # ggplot2::ylim(NA, max(phen_gen$phenotype)*1.5)+
            ggplot2::labs(y= ylab, x = "", title = t) +
            ggplot2::facet_grid(~experiment)
    }
    
    cowplot::ggdraw() +
        cowplot::draw_plot(test, 0, .03, 1, .97) +
        cowplot::draw_label("Chr IV", x = 0.01, y = .08, size = textsize, fontface = "bold", hjust = -.16)+
        ggplot2::geom_rect(ggplot2::aes(xmin = x1+chr, xmax = x1+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2+chr, xmax = x2+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+chr, xmax = x3+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3+blck*.25+chr, xmax = x3+blck*.75-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+chr, xmax = x4+blck-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4+blck*.25+chr, xmax = x4+blck*.75-chr, ymin = y1*3, ymax = y1*3 +width),
                           colour = "black", fill = "blue")+
        cowplot::draw_label("Genome", x = 0.01, y = .05, hjust = -.1, size = textsize, fontface = "bold") +
        ggplot2::geom_rect(ggplot2::aes(xmin = x1, xmax = x1+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x2, xmax = x2+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x3, xmax = x3+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "blue")+
        ggplot2::geom_rect(ggplot2::aes(xmin = x4, xmax = x4+blck, ymin = y2*3, ymax = y2*3 +width),
                           colour = "black", fill = "orange")
}

# More functions for plotting
maxlodplot_kt <- function (map) {
    map1 <- map %>% 
        dplyr::group_by(marker) %>% 
        dplyr::filter(lod ==max(lod))
    tit <- paste0(unique(map$condition), ".", unique(map$trait))
    cis <- map %>% 
        dplyr::group_by(marker) %>% 
        dplyr::mutate(maxlod = max(lod)) %>%
        dplyr::group_by(iteration) %>% 
        dplyr::filter(!is.na(var_exp)) %>%
        dplyr::do(head(., n = 1))
    if (nrow(cis) == 0) {
        plot <- ggplot2::ggplot(map1, ggplot2::aes(x = genotype,y = pheno)) + 
            ggplot2::geom_blank()
        return(plot)
    }
    map1 <- linkagemapping:::cidefiner(cis, map1)
    plot <- ggplot2::ggplot(map1) + ggplot2::aes(x = pos/1e+06,y = lod)
    if (nrow(cis) != 0) {
        plot <- plot + 
            ggplot2::geom_ribbon(ggplot2::aes(x = pos/1e+06,ymin = 0, ymax = ci_lod), fill = "blue", alpha = 0.5) +
            ggplot2::geom_point(data = cis, ggplot2::aes(x = pos/1e+06,y = (1.05 * maxlod)), fill = "red", shape = 25,
                                size = 3.2, show.legend = FALSE) + 
            ggplot2::ylim(0, 1.5*max(cis$maxlod))+
            ggplot2::geom_text(data = cis, ggplot2::aes(x = pos/1e+06, y = (1.2 * maxlod), label = paste0(100 *round(var_exp, digits = 4), "%")), colour = "black",size = 3, hjust = "inward")
    }
    
    plot <- plot + 
        ggplot2::geom_line(size = 0.75, alpha = 0.85) +
        ggplot2::facet_grid(. ~ chr, scales = "free", space = "free") +
        ggplot2::labs(x = "Genomic Position (Mb)", y = "LOD") +
        ggplot2::scale_colour_discrete(name = "Mapping\nIteration") +
        ggplot2::ggtitle(map1$trait[1]) + 
        ggplot2::theme_bw() +
        ggplot2::theme(
            axis.text.x = ggplot2::element_text(size = textsize, color = "black", face = "bold"),
            axis.text.y = ggplot2::element_text(size = textsize, color = "black", face = "bold"),
            axis.title.x = ggplot2::element_text(size = titlesize, color="black", face = "bold"),
            axis.title.y = ggplot2::element_text(size = titlesize,color="black", face = "bold"),
            strip.text.x = ggplot2::element_text(size = titlesize, color = "black"),
            strip.text.y = ggplot2::element_text(size = titlesize, color = "black"),
            strip.background = ggplot2::element_rect(colour = "black", fill = "white", size = 0.75, linetype = "solid"),
            plot.title = ggplot2::element_text(size = (titlesize+5),color="black", face = "bold"),
            panel.background = ggplot2::element_rect(color = "black", size = 1.2)) + 
        ggplot2::labs(title = tit)
    return(plot)
}
pxgplot_kt <- function (cross, map, parent = "N2xCB4856", tit = "", ylab = "") {
    peaks <- map %>% 
        dplyr::group_by(iteration) %>% 
        dplyr::filter(!is.na(var_exp)) %>% 
        dplyr::do(head(., n = 1))
    
    if (nrow(peaks) == 0) {
        stop("No QTL identified")
    }
    uniquemarkers <- gsub("-", "\\.", unique(peaks$marker))
    colnames(cross$pheno) <- gsub("-", "\\.", colnames(cross$pheno))
    pheno <- cross$pheno %>% 
        dplyr::select_(map$trait[1])
    geno <- data.frame(linkagemapping:::extract_genotype(cross)) %>% 
        dplyr::select(which(colnames(.) %in% uniquemarkers)) %>% 
        data.frame(., pheno)
    colnames(geno)[1:(ncol(geno) - 1)] <- sapply(colnames(geno)[1:(ncol(geno) - 1)], 
                                                 function(marker) {paste(unlist(peaks[peaks$marker == gsub("\\.", "-", marker),
                                                                                      c("chr", "pos")]), collapse = ":")})
    colnames(geno)[ncol(geno)] <- "pheno"
    split <- tidyr::gather(geno, marker, genotype, -pheno)
    split$genotype <- sapply(split$genotype, function(x) {
        if (is.na(x)) {
            return(NA)
        }
        if (parent == "N2xCB4856") {
            if (x == -1) {
                "N2"
            }
            else {
                "CB4856"
            }
        }
    })
    split$genotype <- factor(split$genotype, levels = c("N2", "CB4856", "LSJ2", "AF16", "HK104"),
                             labels = c("N2-RIAILs", "CB-RIAILs", "LSJ2-RIAILs", "AF16-RIAILs", "HK104-RIAILs"))
    split <- split[!is.na(split$genotype),]
    test <- ggplot2::ggplot(split) + 
        ggplot2::geom_jitter(ggplot2::aes(x = genotype, y = pheno), alpha = 1, size = 0.5) + 
        ggplot2::geom_boxplot(ggplot2::aes(x = genotype, y = pheno, fill = genotype, alpha = 0.8), outlier.shape = NA) + 
        ggplot2::scale_fill_manual(values = c(`N2-RIAILs` = "orange", `CB-RIAILs` = "blue", LSJ2 = "green", AF16 = "indianred", HK104 = "gold")) + 
        ggplot2::facet_wrap(~marker, ncol = 5) + 
        ggplot2::theme_bw() + 
        ggplot2::theme(axis.text.x = ggplot2::element_text(size = textsize,face = "bold", color = "black"), 
                       axis.text.y = ggplot2::element_text(size = textsize, face = "bold", color = "black"), 
                       axis.title.x = ggplot2::element_text(size = titlesize, face = "bold", color = "black", vjust = -0.3), 
                       axis.title.y = ggplot2::element_text(size = titlesize, face = "bold", color = "black"), 
                       strip.text.x = ggplot2::element_text(size = titlesize, face = "bold", color = "black"), 
                       legend.position = "none", 
                       strip.background = ggplot2::element_rect(fill = "white", colour = "white"),
                       plot.margin = grid::unit(c(-0.5, 0.2, 0.2, 0.2), "cm"),
                       panel.background = ggplot2::element_rect(color = "black",size = 1.2)) + 
        ggplot2::labs(x = "", y = ylab, title = "")
    
    cowplot::ggdraw() +
        cowplot::draw_plot(test, 0, .04, 1, .96)
}
plot_add_int <- function(df, tit = "") {
    ggplot2::ggplot(df) +
        ggplot2::geom_bar(data = df, ggplot2::aes(x = trait, y = VE, fill = factor(type, levels = c("Interactive", "Additive"))),
                          stat = "identity") +
        ggplot2::scale_fill_manual("Key", values = c("slateblue2", "cadetblue3"))+
        ggplot2::labs(y = "Variance Explained", x = tit) + 
        ggplot2::geom_errorbar(ggplot2::aes(x = trait, ymin = lower, ymax = higher, group = n, linetype = type),
                               width = 0.7, position = position_dodge(0.4)) +
        ggplot2::theme(axis.text.x = ggplot2::element_blank(),
                       axis.text.y = ggplot2::element_text(size = textsize),
                       axis.ticks.x = ggplot2::element_blank(),
                       axis.title.x = ggplot2::element_text(size = titlesize, face = "bold"),
                       axis.title.y = ggplot2::element_text(size = titlesize, face = "bold"),
                       legend.position = "none")
}
getpheno <- function(nil, exp, cond, trt) {
    regressed <- allNILCSSregressed %>%
        dplyr::filter(condition == cond, trait == trt, experiment == exp, strain == nil)
    pheno <- abs(max(regressed$phenotype))
    range <- (max(regressed$phenotype)) + (min(regressed$phenotype))
    if(range < 65 && range > 5) { newpheno <- pheno + 7 } else { newpheno <- (pheno*1.4) } 
}
plotstats <- function(drug, trt, expr) {
    fullstats <- NULL
    for(j in 1:length(expr)) {
        e <- expr[j]
        stats <- css_nil_stats %>%
            dplyr::filter(exp == e, condition == drug, trait == trt)
        
        #make an empty dataframe
        statsdf <- setNames(data.frame(matrix(ncol = 4, nrow = 4)), c("strain", "significant", "pheno", "experiment"))
        statsdf$strain <- c("N2", "CB4856", stats$N2nil, stats$CBnil)
        statsdf$experiment <- c(e)
        
        sig <- 0.05
        
        parents <- if(stats$par_sig < 0.05) {parents <- TRUE} else {parents <- FALSE}
        nils <- if(stats$nils_sig< 0.05) {nils <- TRUE} else {nils <- FALSE}
        n2nilcb <- if(stats$N2nil_sig_CB< 0.05) {n2nilcb <- TRUE} else {n2nilcb <- FALSE}
        n2niln2 <- if(stats$N2nil_sig_N2< 0.05) {n2niln2 <- TRUE} else {n2niln2 <- FALSE}
        cbnilcb <- if(stats$CBnil_sig_CB< 0.05) {cbnilcb <- TRUE} else {cbnilcb <- FALSE}
        cbniln2 <- if(stats$CBnil_sig_N2< 0.05) {cbniln2 <- TRUE} else {cbniln2 <- FALSE}
        
        #If parents are sig
        if(parents==T) {
            #Check if nils are sig
            if(nils==T) {
                if(n2nilcb==F && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "b"
                        CBnilval <- "a"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "bc"
                        N2nilval <- "c"
                        CBnilval <- "ab"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "c"
                        N2nilval <- "ac"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ac"
                        CBval <- "BD"
                        N2nilval <- "ab"
                        CBnilval <- "CD"
                    }
                } else if(n2nilcb==T && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "d"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "a"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                } else if(n2nilcb==T && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "a"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "ab"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "c"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "c"
                        N2nilval <- "a"
                        CBnilval <- "bc"
                    }
                } else if(n2nilcb==F && cbniln2==T) { 
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "bc"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "ab"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "bc"
                        N2nilval <- "ab"
                        CBnilval <- "c"
                    }
                }
                #nils not sig
            } else  {
                if(n2nilcb==F && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "bc"
                        CBnilval <- "ac"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "b"
                        CBnilval <- "ab"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "bc"
                        N2nilval <- "ab"
                        CBnilval <- "a"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "ab"
                        CBnilval <- "ab"
                    }
                } else if(n2nilcb==T && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "bc"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "ac"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "ac"
                        CBnilval <- "bc"
                    }
                } else if(n2nilcb==T && cbniln2==F) { 
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "ac"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "c"
                        CBnilval <- "abc"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "a"
                        CBnilval <- "a"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "a"
                        CBnilval <- "ab"
                    }
                } else if(n2nilcb==F && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "bc"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "b"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "abc"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "b"
                        N2nilval <- "ab"
                        CBnilval <- "b"
                    }
                }
                #parents are ns
            } 
        } else  {
            #check if nils are sig
            if(nils==T) {
                if(n2nilcb==F && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "ac"
                        N2nilval <- "c"
                        CBnilval <- "b"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "b"
                        CBnilval <- "a"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "ab"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                } else if(n2nilcb==T && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "c"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "ac"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                } else if(n2nilcb==T && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "c"
                        CBnilval <- "b"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "a"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "abc"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "a"
                    }
                } else if(n2nilcb==F && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "abc"
                        N2nilval <- "b"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "a"
                        CBnilval <- "b"
                    }
                }
                #nils not sig 
            } else {
                if(n2nilcb==F && cbniln2==F) {
                    #START HERE
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "bc"
                        CBnilval <- "ac"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "b"
                        CBnilval <- "ab"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "ac"
                        CBnilval <- "bc"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "a"
                        CBnilval <- "a"
                    }
                } else if(n2nilcb==T && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "c"
                        CBnilval <- "bc"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "c"
                        CBnilval <- "bc"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "bc"
                        CBnilval <- "c"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "ac"
                        N2nilval <- "BD"
                        CBnilval <- "CD"
                    }
                } else if(n2nilcb==T && cbniln2==F) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "c"
                        CBnilval <- "bc"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "ab"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "ab"
                        CBval <- "a"
                        N2nilval <- "b"
                        CBnilval <- "ab"
                    }
                } else if(n2nilcb==F && cbniln2==T) {
                    if(n2niln2==T && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "bc"
                        CBnilval <- "c"
                    }
                    if(n2niln2==T && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "b"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==T) {
                        N2val <- "a"
                        CBval <- "a"
                        N2nilval <- "ab"
                        CBnilval <- "b"
                    }
                    if(n2niln2==F && cbnilcb==F) {
                        N2val <- "a"
                        CBval <- "ab"
                        N2nilval <- "ac"
                        CBnilval <- "bc"
                    }
                }
            }
        }
        statsdf$significant <- c(N2val, CBval, N2nilval, CBnilval)
        statsdf$pheno <- sapply(statsdf$strain, function(x) getpheno(x, e, drug, trt))
        #Bind stats dataframes
        fullstats <- rbind(fullstats, statsdf)
    }
    return(fullstats)
}

splitmap <- allAnnotatedLods %>%
    dplyr::mutate(condition = stringr::str_split_fixed(trait, "\\.", 2)[,1]) %>%
    dplyr::mutate(trait = stringr::str_split_fixed(trait, "\\.", 2)[,2]) %>%
    dplyr::arrange(condition, trait)


```

```{r s3_2,fig.width=7.5, fig.height=6,warning=F, message=F, echo=F,comment="K",results='asis', dpi = 300}

for(drug in unique(splitmap$condition)) {
    drugmap <- splitmap %>%
        dplyr::filter(condition == drug) %>%
        na.omit() %>%
        dplyr::distinct(marker, trait, .keep_all = TRUE)
    
    for(trt in unique(drugmap$trait)) {
        traitmap <- drugmap %>% 
            dplyr::filter(trait == trt) %>%
            dplyr::arrange(chr, pos)
        
        #Mapping
        lod <- maxlodplot_kt(splitmap %>% dplyr::filter(condition == drug, trait == trt))
        VE <- plot_add_int(VE_SE_QTL %>% dplyr::filter(condition == drug, trait == trt))
        
        # make blank plots up to 5 for RIAIL phenotype split so they are all even
        ril <- pxgplot_kt(phenocross, allAnnotatedLods %>% dplyr::filter(trait == paste0(drug, ".", trt)))
        
        if(nrow(traitmap) < 5) {
            ril_plots <- cowplot::plot_grid(ril, rep(NULL, 5-nrow(traitmap)), rel_widths = c(nrow(traitmap), rep(1, 5-nrow(traitmap))))
        } else {
            ril_plots <- cowplot::plot_grid(ril)
        }
        # print figures
        lod_ve <- cowplot::plot_grid(lod, VE, nrow = 1, ncol = 2, rel_widths = c(5, 1), labels = c("A", "B"))
        print(cowplot::plot_grid(lod_ve, ril_plots, nrow = 2, ncol = 1, labels = c("", "C")))

    }
}


```

**Figure S3** Linkage mapping results and variance component contributions for 47 principal components with a signficant QTL. (**A**) The maximum LOD score (y-axis) for each marker across the genome (x-axis), split by chromosome, is plotted as a black line. Significant QTL are indicated by a red triangle at the marker with the maximum LOD score and a light blue ribbon spanning the 95% confidence interval. The estimated phenotypic variation in the RIAILs explained by each QTL is indicated above each peak. (**B**) The proportion of phenotypic variation across the RIAILs caused by additive (light blue) and interactive (dark blue) genetic components are indicated. Solid and dashed error bars show the standard error around additive and interactive component estimates, respectively. (**C**) For each QTL plotted in A, the residual phenotypes (y-axis) of RIAILs split by genotype at the marker with the maximum LOD score (x-axis) are plotted as Tukey box plots. Data from RIAILs with the N2 or the CB4856 allele at the QTL peak marker are colored in orange and blue, respectively. 


```{r s4,warning=F, message=F, echo=F,comment="K",results='asis',fig.width=7.5, fig.height=8}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(cowplot)
library(Hmisc)
# source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS4_allAnnotatedLods.Rda?raw=TRUE")
# source_data("https://github.com/AndersenLab/QTLhotspot/blob/master/data/FileS6_varianceComponents.Rda?raw=TRUE")

peaks <- allAnnotatedLods %>%
    na.omit()

VE <- varianceComponents %>%
    dplyr::select(condition, trait, narrowsense_h2, interaction_VE) %>%
    tidyr::gather(type, VE, narrowsense_h2:interaction_VE) %>%
    dplyr::mutate(type = ifelse(type == "narrowsense_h2", "Additive", "Interactive"))

SE <- varianceComponents %>%
    dplyr::select(condition, trait, narrowsense_h2_SE, interaction_SE) %>%
    tidyr::gather(type, SE, narrowsense_h2_SE:interaction_SE) %>%
    dplyr::mutate(type = ifelse(type == "narrowsense_h2_SE", "Additive", "Interactive"))

VE_SE_QTL <- dplyr::left_join(VE, SE) %>%
    dplyr::mutate(n = 1:nrow(.))

VE_SE_QTL$type <- factor(VE_SE_QTL$type, levels = c("Interactive", "Additive"))
VE_SE_QTL <- VE_SE_QTL %>%
    mutate(lower = VE - SE, higher = VE + SE)

VE_SE_QTL$lower[VE_SE_QTL$type == "Interactive"] <- with(VE_SE_QTL,VE[type == "Additive"] - SE[type == "Interactive"] + VE[type == "Interactive"])
VE_SE_QTL$higher[VE_SE_QTL$type == "Interactive"] <- with(VE_SE_QTL,VE[type == "Interactive"] + VE[type == "Additive"] + SE[type == "Interactive"])

for(i in 1:nrow(VE_SE_QTL)) {
    if(VE_SE_QTL$lower[i] < 0) {
        VE_SE_QTL$lower[i] <- 0
    }
}

textsize <- 8
titlesize <- 10

VE_SE_QTL <- VE_SE_QTL %>%
    dplyr::mutate(condtrt = paste0(condition, ".", trait)) %>%
    dplyr::filter(condtrt %in% unique(peaks$trait)) %>%
    dplyr::arrange(condition, trait) %>%
    dplyr::mutate(condition = Hmisc::capitalize(condition))

allinteractions <- ggplot2::ggplot(VE_SE_QTL) +
    ggplot2::geom_bar(data = VE_SE_QTL,ggplot2:: aes(x = trait, 
                                                     y = VE, fill = factor(type, levels = c("Interactive", "Additive"))), stat = "identity") +
    ggplot2::scale_fill_manual("Key", values = c("slateblue2", "cadetblue3"))+
    ggplot2::coord_flip() +
    ggplot2::facet_grid(~condition, scales = "free_y", switch = "both") +
    ggplot2::labs(y = "Variance Explained", x = "") + 
    ggplot2::geom_errorbar(ggplot2::aes(x = trait, ymin = lower, ymax = higher, group = n, linetype = type), width = 0.7, position = position_dodge(0.4)) +
    ggplot2::theme(legend.position = "none",
                   strip.background = ggplot2::element_rect(colour = "black", fill = "white", linetype = "solid", size = 1),
                   strip.text = ggplot2::element_text(size = titlesize, face = "bold"),
                   strip.text.x = ggplot2::element_text(margin = margin(4, 0, 4, 0)),
                   axis.text.x = ggplot2::element_text(size = textsize),
                   axis.text.y = ggplot2::element_text(size = textsize),
                   axis.title.x = ggplot2::element_blank()) +
    ggplot2::facet_wrap(~condition, scales = "free", ncol = 3, nrow = 6)

legstartx <- 0.5 #position where the top left corner of the legend begins
legstarty <- 0.1 #position where the top left corner of the legend begins
width <- 0.02 #width of the boxes/lines of legend
sp <- 0.01 #spacing within legend

plot <- cowplot::ggdraw() +
    cowplot::draw_plot(allinteractions, 0, 0, 1, 1) +
    ggplot2::geom_rect(ggplot2::aes(xmin = legstartx - sp, xmax = legstartx + width*12+sp*2, 
                                    ymin = legstarty-width-sp, ymax = legstarty + sp), colour = "black", fill = "white") +
    ggplot2::geom_rect(ggplot2::aes(xmin = legstartx, xmax = legstartx + width, ymax = legstarty, ymin = legstarty - width*0.8), 
                       colour = "black", fill = "cadetblue3") +
    cowplot::draw_label("Additive", x = legstartx + width*3 + sp, y = legstarty - width*0.35, size = titlesize, fontface = "bold") +
    ggplot2::geom_segment(ggplot2::aes(x = legstartx - width*.25, xend = legstartx + width*1.35, 
                                       y = legstarty - width*.35, yend = legstarty - width*.35), colour = "black", linetype = "solid")+ 
    ggplot2::geom_rect(ggplot2::aes(xmin = legstartx+width*6, xmax = legstartx + width*7, ymax = legstarty, 
                                    ymin = legstarty - width*0.8), colour = "black", fill = "slateblue2") +
    cowplot::draw_label("Interactive", x = legstartx + width*9.5 + sp, y = legstarty - width*0.35, size = titlesize, fontface = "bold") +
    ggplot2::geom_segment(ggplot2::aes(x = legstartx+width*5.75, xend = legstartx + width*7.5, 
                                       y = legstarty - width*.35, yend = legstarty - width*.35), colour = "black", linetype = "dashed") 

print(plot)

```

**Figure S4** Additive and interactive genetic components. The proportion of phenotypic variation predicted to be caused by additive and interactive genetic components is shown for each of the 47 traits that represent the 82 QTL identified by linkage mapping. For each toxin, the fraction of phenotypic variation (x-axis) in a given principal component (y-axis) that is attributable to additive (light blue) versus interactive (dark blue) genetic components is shown as a stacked bar plot. Solid and dashed error bars show the standard error around the mean of additive and interactive components, respectively.

```{r s5,warning = F, message = F, echo = F, fig.width = 7.5, fig.height = 3, comment = "K", results = "asis"}
library(tidyverse)

# abbreviate category names for plotting
plotcat <- assays_category %>%
    dplyr::mutate(primary_category = case_when(primary_category == "recapitulation" ~ "Recapitulation",
                                               primary_category == "no_qtl_effect" ~ "No effect",
                                               primary_category == "unidirectional_transgressive" ~ "Unidirectional",
                                               primary_category == "bidirectional_transgressive" ~ "Bidirectional",
                                               primary_category == "miscellaneous" ~ "Miscellaneous",
                                               primary_category == "parents_not_significant" ~ "No parental diff."))

# filter to only keep traits that mapped to hotspot that we tested in NILs
# filter to only keep traits that mapped to hotspot that we tested in NILs
hotspots <- allAnnotatedLods %>%
    na.omit() %>%
    dplyr::mutate(HS_IVL = ifelse(chr == "IV" & ci_l_pos <  9334865 & ci_r_pos > 3684741, T, F),
                  HS_IVR = ifelse(chr == "IV" & ci_l_pos < 17493829 & ci_r_pos > 12865211, T, F),
                  HS_V = ifelse(chr == "V" & ci_l_pos < 13839858 & ci_r_pos > 7082839, T, F)) %>%
    dplyr::filter(HS_IVL | HS_IVR | HS_V) %>%
    dplyr::mutate(condition = stringr::str_split_fixed(trait, "\\.", 2)[,1]) %>%
    dplyr::mutate(NIL_test = ifelse(HS_IVL & condition %in% c("silver", "carmustine", "chlorothalonil", "tunicamycin", "cisplatin"), T,
                                    ifelse(HS_IVR & condition %in% c("diquat", "irinotecan", "chlorothalonil", "fluoxetine", "mechlorethamine"), T, 
                                           ifelse(HS_V & condition %in% c("cisplatin", "carmustine", "chlorothalonil", "paraquat", "silver"), T, F)))) %>%
    dplyr::filter(NIL_test == T) %>%
    dplyr::arrange(condition, trait)

out <- NULL
for(drug in unique(hotspots$condition)) {
    drugmap <- hotspots %>%
        dplyr::filter(condition == drug) %>%
        na.omit() %>%
        dplyr::distinct(marker, trait, .keep_all = TRUE)
    
    for(trt in unique(drugmap$trait)) {
        out = c(out, knit_child('~/Dropbox/AndersenLab/QTLpaper/git_repo/QTLhotspot/scripts/figureS5_template.Rmd'))
    }
}
```

`r paste(knit(text = out), collapse = '\n')`

**Figure S5** NIL and CSS phenotypes. For traits in which NILs and CSSs were assayed, the residual phenotypic values (y-axis) of N2, CB4856, and reciprocal introgressed NILs or CSSs are plotted as Tukey box plots. Different letters represent significant differences (p<0.05) between strains (TukeyHSD). The genotype of each strain on the x-axis is modeled by the colored rectangles beneath the plots (N2 genotypes are orange, CB4856 genotypes are blue). The category assigned to each test of recapitulation of QTL effect is marked above each NIL/CSS phenotype plot. Traits are grouped by the principal component they were selected to represent, based on correlation clusters from Figure S2.

```{r s6, echo=FALSE, out.width='100%'}
figS5_path <- "https://cdn.rawgit.com/AndersenLab/QTLhotspot/ed981acd/figures/extra/figS5_categorization2.pdf"
download.file(figS5_path, destfile = "~/Downloads/figS5.pdf")
knitr::include_graphics("~/Downloads/figS5.pdf")

```

**Figure S6** NIL and CSS trait categorizations. (**A**) Flowchart for categorizing traits from the NIL or CSS tests of recapitulation of QTL effects. (**B**) Six potential categories for chromosome V traits tested in both NIL and CSS assays with a significant and consistent parental phenotpic split across both assays. The miscellaneous category is not depicted, but encompasses any other combination of NIL and CSS assay results.

```{r s7,warning=F, message=F, echo=F,comment="K",results='asis',fig.width=7.5, fig.height=6}

# Look at all traits that are grouped for a QTL and see if there are correlations
# load("~/Dropbox/AndersenLab/QTLpaper/git_repo/QTLhotspot/data/FileS8_PC_trait_correlation.Rda")

# 18 traits in hotspots that were tested with NILs
niltest <- PC_trait_correlation %>%
    dplyr::filter(NIL_test == T, cluster_traits == T)

assays_category <- assays_category %>%
    tidyr::unite(trait, c("condition", "trait"), sep = ".")

cortrtcat <- left_join(niltest, assays_category) %>%
    dplyr::filter(exp != "CSSV") %>%
    dplyr::distinct(PC_trait, trait, exp, .keep_all = T) %>%
    dplyr::group_by(PC_trait, exp) %>%
    dplyr::mutate(num_traits = n()) %>%
    dplyr::group_by(PC_trait, exp, primary_category) %>%
    dplyr::mutate(num_cat = n()) %>%
    dplyr::mutate(perc_cat = num_cat / num_traits *100) %>%
    dplyr::arrange(desc(PC_trait), desc(exp)) %>%
    dplyr::distinct(PC_trait, exp, primary_category, .keep_all = T)

cortrtcat$PC_trait <- factor(cortrtcat$PC_trait, levels = unique(cortrtcat$PC_trait))
cortrtcat$primary_category <- factor(cortrtcat$primary_category, levels = c("parents_not_significant", "miscellaneous", "bidirectional_transgressive", "unidirectional_transgressive", "no_qtl_effect", "recapitulation"), labels = c("Parents Not Significant", "Miscellaneous", "Bidirectional Transgressive",
                                                                                                                                                                                                                                      "Unidirectional Transgressive", "No Qtl Effect", "Recapitulation"))

# stacked barchart
corgroupplot <- ggplot2::ggplot(data = cortrtcat, ggplot2::aes(x = PC_trait, y = perc_cat)) +
    ggplot2::geom_bar(aes(fill = primary_category), stat = "identity") +
    ggplot2::scale_fill_manual(values = c(Recapitulation = "forestgreen", `No Qtl Effect` = "darkred", `Unidirectional Transgressive` = "steelblue1",
                                          `Bidirectional Transgressive` = "steelblue4", `Miscellaneous` = "grey75", `Parents Not Significant` = "grey51"),
                               name = "Category", limits = c("Recapitulation", "No Qtl Effect", "Unidirectional Transgressive", "Bidirectional Transgressive",
                                                             "Miscellaneous", "Parents Not Significant"))+
    ggplot2::geom_text(ggplot2::aes(label = num_traits, y = 95), color = "white") +
    ggplot2::theme_bw() +
    ggplot2::theme(axis.text.x = ggplot2::element_text(vjust = 0.5), 
                   axis.text = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                   axis.title = ggplot2::element_text(size = 12, face = "bold", color = "black")) +
    ggplot2::labs(y = "Percent of traits", x = "Principal Component") +
    ggplot2::coord_flip() + 
    ggplot2::facet_grid(factor(exp)~., scales = "free", space = "free")
print(corgroupplot)

```

**Figure S7** Categorization of correlated traits. The y-axis shows the principal component tested in NIL assays, split by hotspot. Each color represents the NIL assay categorization for each trait within a correlation cluster - either recapitulation (green), no QTL effect (red), unidirectional transgressive phenotype (light blue), bidirectional transgressive phenotype (dark blue), miscellaneous (light grey) or no significant parental difference (dark grey). The percent of all traits within the correlation cluster for each principal component that falls within a given category is shown on the y-axis. Numbers on the right of each bar indicate the number of traits within a correlation cluster.
